apps/openmw/mwworld/projectilemanager.cpp

```diff
@@ -406,29 +406,27 @@ namespace MWWorld
             // TODO: use a proper btRigidBody / btGhostObject?
             MWPhysics::PhysicsSystem::RayResult result = mPhysics->castRay(pos, newPos, caster, 0xff, MWPhysics::CollisionType_Projectile);
 
+            bool hitDeadActor = !result.mHitObject.isEmpty() && result.mHitObject.getClass().isActor() && result.mHitObject.getClass().getCreatureStats(result.mHitObject).isDead();
```

Comment:
The collision mask of the Actor already excludes CollisionType_Projectile when the actor is dead, see https://github.com/OpenMW/openmw/blob/master/apps/openmw/mwphysics/actor.cpp#L88). I think you just need to change the mask for the castRay to CollisionType_Projectile instead of 0xff and all should work as intended.